### PR TITLE
[Marketo Static Lists] Enhancements

### PR DESCRIPTION
**Pull request by maryamsharif**
[STRATCONN-3443](https://segment.atlassian.net/jira/software/c/projects/STRATCONN/boards/310?assignee=60873be283b8c6006bcfb3a0&assignee=620ae2beef2bd5006ae88efb&selectedIssue=STRATCONN-3443) - Post testing enhancements.

[Datadog Notebook](https://segment.datadoghq.com/notebook/7111156/marketo-action-destination?range=2592000000&tpl_var_env=stage&start=1701021085341&live=true).
## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

[STRATCONN-3443]: https://segment.atlassian.net/browse/STRATCONN-3443?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/destination-actions/src/destinations/marketo-static-lists/addToList/index.ts [hunk 1]
```diff
@@ -15,11 +15,13 @@ const action: ActionDefinition<Settings, Payload> = {
     batch_size: { ...batch_size },
     event_name: { ...event_name }
   },
-  perform: async (request, { settings, payload }) => {
-    return addToList(request, settings, [payload])
+  perform: async (request, { settings, payload, statsContext }) => {
+    statsContext?.statsClient?.incr('addToList', 1, statsContext?.tags)
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/marketo-static-lists/addToList/index.ts:19]: Nice. I really like how we've set up observability for this type on action. However, I think we might need to standardize the name of the metric so we can create generic widgets in DD. So far, we have used `addToAudience` and `removeFromAudience` so I'd stick to those. If you are keeping monitoring for this inside a notebook and feel like the current monitors could detect failures then let's keep it as is.

File: packages/destination-actions/src/destinations/marketo-static-lists/functions.ts [hunk 2]
```diff
@@ -32,16 +42,25 @@ export async function addToList(request: RequestClient, settings: Settings, payl
...
+export async function removeFromList(
+  request: RequestClient,
+  settings: Settings,
+  payloads: RemoveFromListPayload[],
+  statsContext?: StatsContext
+) {
+  // Keep only the scheme and host from the endpoint
+  // Marketo shows endpoint with trailing "/rest", which we don't want
+  const api_endpoint = settings.api_endpoint.replace('/rest', '')
+  const emailsToRemove = extractEmails(payloads, ',')
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/marketo-static-lists/functions.ts:61]: No string interpolation here?

**Review by marinhero — APPROVED** [review]

File: packages/destination-actions/src/destinations/marketo-static-lists/functions.ts (hunk 2 above)
Comment (maryamsharif) [reply packages/destination-actions/src/destinations/marketo-static-lists/functions.ts:61]: When removing emails from a list, we just need a string of emails separated by commas.

File: packages/destination-actions/src/destinations/marketo-static-lists/addToList/index.ts (hunk 1 above)
Comment (maryamsharif) [reply packages/destination-actions/src/destinations/marketo-static-lists/addToList/index.ts:19]: That makes sense. I will make the change to standardize this.

**Comment by joe-ayoub-segment** [comment]
Will deploy and push Wednesday 10th Jan.

**Comment by joe-ayoub-segment** [comment]
Hi @maryamsharif this PR has been deployed

**Pull request merged**